File: Data Structures & Algorithms/is-anagram/submission-2.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char, int> m;
        for(auto x : s) {
            if(m.count(x)) {
                m[x]++;
            } else {
                m.insert({x, 1});
            }
        }
        for(auto x : t) {
            if(m.count(x)) {
                if(m[x] - 1 >= 0) {
                    m[x]--;
                } else {
                    return false;
                }
            } else {
                return false;
            }
        }
        int sum = 0;
        for(auto x : m) {
            sum += x.second; 
        }
        return sum ? false : true;
    }
};

File: Data Structures & Algorithms/counting-bits/submission-2.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        ret = [0] * (n+1)

        for i in range(n+1):
            count = 0
            for j in range(32):
                if i & (1 << j) != 0:
                    count += 1
            ret[i] = count
        return ret